backend/api/communication/src/test/java/co/airy/core/api/communication/WebSocketControllerTest.java

```diff
@@ -135,17 +133,17 @@ void init() throws Exception {
 
     @Test
     void sendsToWebsocket() throws Exception {
-        final CompletableFuture<MessageUpsertPayload> messageFuture = subscribe(port, MessageUpsertPayload.class, QUEUE_MESSAGE, jwt);
+        final CompletableFuture<Map> messageFuture = subscribe(port, Map.class, QUEUE_MESSAGE, jwt);
```

Comment:
cc @paulodiniz and @lucapette The tests were failing, because we cannot parse the message content back with Jackson. If we want to be able to do that we would have to add [Jackson polymorphic type annotations](https://github.com/FasterXML/jackson-docs/wiki/JacksonPolymorphicDeserialization) to the content model, which would mean that message content would by default be returned with a `type` field.

Perphaps we have to this anyway, because otherwise, clients will also not know how to deserialize these fields. What do you think?

Reply:
Going the Jackson polymorphism + generating Typescript definitions way would yield something like this: https://jechlin.github.io/ts-gen-aws/

Reply:
@chrismatix @lucapette I like and agree with the idea of adding a `type` to the payload. It *immediately* brought the `json-api` spec to mind.

https://jsonapi.org/

What do you think about trying to go full `json-api` spec by making the response in this format?

```json5
{
  "data": [{
     "type": ":conversation"
     ...
  }],
  
}
```

Reply:
As @chrismatix pointed, changing to `json-api` would demand changing the whole api.

On the `type` attribute, I like it 👍

Reply:
same, a `type` field can come handy in a many situations